OpenRA.Mods.Common/Activities/Rearm.cs

```diff
@@ -56,7 +56,8 @@ public override Activity Tick(Actor self)
 				if (!pool.GiveAmmo())
 					continue;
 
-				hostBuilding.Trait<RenderBuilding>().PlayCustomAnim(hostBuilding, "active");
+				if (hostBuilding.HasTrait<WithSpriteBody>())
```

Comment:
This is an unnecessary trait lookup. You are already doing a null check in line 53.